testing/unit_testing/validate_cohorts_test/test_ValidateCohorts.pf

```diff
@@ -0,0 +1,145 @@
+      class(TestValidateCohorts), intent(inout) :: this           ! test object
+      type(fates_patch_type),     pointer       :: patch          ! patch object
+      type(fates_cohort_type),    pointer       :: new_node, head ! cohort objects
+      
+      ! heights for cohors
+      real(r8), parameter :: heights(8) = (/2.0_r8, 5.0_r8, 10.0_r8, 12.0_r8, 12.5_r8,   &
+         15.0_r8, 20.0_r8, 25.0_r8/)
+
+      ! hard-code a linked list
+      allocate(patch)
+      
+      allocate(head)
+      patch%shortest => head
+      head%height = heights(1)
+      
+      allocate(new_node)
+      new_node%height = heights(2)
+      head%taller => new_node
+      new_node%shorter => head
+      
+      allocate(new_node)
+      new_node%height = heights(3)
+      head%taller%taller => new_node
+      new_node%shorter => head%taller
+      
+      allocate(new_node)
+      new_node%height = heights(4)
+      head%taller%taller%taller => new_node
+      new_node%shorter => head%taller%taller
+      
+      allocate(new_node)
+      new_node%height = heights(5)
+      head%taller%taller%taller%taller => new_node
+      new_node%shorter => head%taller%taller%taller
+      
+      allocate(new_node)
+      new_node%height = heights(6)
+      head%taller%taller%taller%taller%taller => new_node
+      new_node%shorter => head%taller%taller%taller%taller
+      
+      allocate(new_node)
+      new_node%height = heights(7)
+      head%taller%taller%taller%taller%taller%taller => new_node
+      new_node%shorter => head%taller%taller%taller%taller%taller
+      
+      allocate(new_node)
+      new_node%height = heights(8)
+      head%taller%taller%taller%taller%taller%taller%taller => new_node
+      new_node%shorter => head%taller%taller%taller%taller%taller%taller
+      patch%tallest => new_node
+      
+      ! should pass
+      call patch%ValidateCohorts()
+      
+    end subroutine MultipleCohorts_ValidateCohorts_Returns
+    
+    @Test 
+    subroutine BrokenList_ValidateCohorts_Errors(this)
+      ! tests that for a patch with a broken chain the subroutine errors correctly
+      class(TestValidateCohorts), intent(inout) :: this                      ! test object
+      type(fates_patch_type),     pointer       :: patch                     ! patch object
+      type(fates_cohort_type),    pointer       :: cohort1, cohort2, cohort3 ! cohorts
+      
+      ! create a new patch
+      allocate(patch)
+
+      ! allocate and link cohorts
+      allocate(cohort1)
+      allocate(cohort2)
+      allocate(cohort3)
+      
+      ! set up a broken forward link
+      patch%shortest => cohort1
+      patch%tallest => cohort3
+      
+      cohort1%taller => cohort2
+      cohort2%shorter => cohort1
+      cohort2%taller => cohort3
+      cohort3%shorter => null()  ! break the forward link
+      
+      ! should fail
+      call patch%ValidateCohorts()
```

Comment:
I'm thinking there should be an `assertExceptionRaised` call here?

Reply:
> oh weird... yes will update that

And if this test currently passes without that `assertExceptionRaised`, then it's probably worth figuring out why (because it means that other tests may be passing when they shouldn't).

Reply:
actually I do see this here now. I think perhaps I had added that while you were in the process of reviewing? sorry about that.

Reply:
Ah yes, I see it now - I think you're right about what happened (and now GitHub moved this comment to the wrong place in the file, too). No worries.